### ISSUE TITLE
Improve Critical error message related to configuration issues

### ISSUE THREAD

**Issue by ppf2** [elastic/connectors#763]
Running 8.7.0.0 of the connector service.

If there's a mysql configuration where the port is left empty:

<img width="88" alt="image" src="https://user-images.githubusercontent.com/7216393/231606837-df4de1ae-169e-45df-bfc0-5671389035f6.png">

It throws a critical error with no details on where the problem field is located:

```
[FMWK][16:12:37][INFO] Preflight checks...
[FMWK][16:12:37][INFO] Waiting for NodeConfig(scheme='https', host='localhost', port=9200, path_prefix='', headers={}, connections_per_node=10, request_timeout=10.0, http_compress=False, verify_certs=True, ca_certs=None, client_cert=None, client_key=None, ssl_assert_hostname=None, ssl_assert_fingerprint=None, ssl_version=None, ssl_context=None, ssl_show_warn=True, _extras={}) (so far: 0 secs)
[FMWK][16:12:37][INFO] Service started, listening to events from https://localhost:9200
[FMWK][16:12:37][INFO] Service started, listening to events from https://localhost:9200
[FMWK][16:12:37][INFO] Successfully started Job cleanup task...
[FMWK][16:12:37][INFO] Start cleaning up orphaned jobs...
[FMWK][16:12:38][CRITICAL] invalid literal for int() with base 10: ''
Traceback (most recent call last):
  File "<path>/connectors-python/connectors/services/job_scheduling.py", line 129, in _run
    await self._schedule(connector)
  File "<path>/connectors-python/connectors/services/job_scheduling.py", line 95, in _schedule
    validator=source_klass(connector.configuration)
  File "<path>/connectors-python/connectors/sources/mysql.py", line 275, in __init__
    super().__init__(configuration=configuration)
  File "<path>/connectors-python/connectors/source.py", line 350, in __init__
    self.configuration.set_defaults(self.get_default_configuration())
  File "<path>/connectors-python/connectors/source.py", line 244, in set_defaults
    self._config[name].type = item["type"]
  File "<path>/connectors-python/connectors/source.py", line 88, in type
    self.value = self._convert(self.value, self._type)
  File "<path>/connectors-python/connectors/source.py", line 114, in _convert
    return int(value)
ValueError: invalid literal for int() with base 10: ''
[FMWK][16:12:38][INFO] No orphaned jobs found. Skipping...
[FMWK][16:12:38][INFO] Start cleaning up idle jobs...
[FMWK][16:12:39][INFO] No idle jobs found. Skipping...
```

I can only deduce from the stack trace above that it is a mySQL connection and is probably the port field since it's complaining about an `int` 😄 Imagine a user has tens or more mySQL search indices?  It can be different to quickly identify the problem search index.

A better experience will be to include the following information in the exception message:
- Name of the search-index with the configuration problem
- Name of the configuration field that is causing the exception.

We might want to consider using a different logging level for these messages.  The connector service still functions properly with this error related to a misconfigured connector (it can run other sync jobs fine).  So perhaps we can use something like `ERROR` instead of `CRITICAL`?

**Comment by artem-shelkovnikov** [comment]
Afaik it's fixed now @navarone-feekery - is it the case?

**Comment by navarone-feekery** [comment]
Yes this is fixed from 8.8 by [the rich configurable fields work](https://github.com/elastic/enterprise-search-team/issues/3065) we did. Missing or invalid fields are now validated and fail gracefully. For example in this situation the user would receive a `ConfigurableFieldValueError` error with `'port' cannot be empty`. This error would still occur in 8.7, however.

**Comment by navarone-feekery** [comment]
Thanks for bringing it our attention in any case.

Because this is fixed in 8.8+ and [the connector was considered to be in beta for 8.7](https://www.elastic.co/guide/en/enterprise-search/8.7/connectors-mysql.html#connectors-mysql-prerequisites), I don't think this is worth fixing for 8.7. I'm going to close the issue.
